Practical-6/user-list-app/src/reducer/userListReducer.js

```diff
@@ -0,0 +1,71 @@
+const initialData = {
+  userDetails: [],
+  isLoading: false,
+  error: '',
+  userProfile: [],
+  pagination: 1,
+}
+const userListReducers = (state = initialData, action) => {
+  switch (action.type) {
+    case 'REMOVE_USER':
+      return {
+        userDetails: [...state.userDetails.filter((user) => user.id!== action.payload.id)],
```

Comment:
you can directly use state.userDetails.filter. No need use spread operator if you not copied previous state

Reply:
It's necessary to spread the previous state

Reply:
then it should be like this
 return {
         ...state,
        userDetails: [state.userDetails.filter((user) => user.id!== action.payload.id)]
}